internal/provider/providers/aws/provider.go

```diff
@@ -0,0 +1,184 @@
+	}
+
+	err = validateSettings(providerSpecificSettings, domain, host)
+	if err != nil {
+		return nil, fmt.Errorf("validating provider specific settings: %w", err)
+	}
+	ttl := int64(300)
+	if providerSpecificSettings.TTL != nil {
+		ttl = *providerSpecificSettings.TTL
+	}
+
+	region := "us-east-1"
+	if providerSpecificSettings.Region != "" {
+		region = providerSpecificSettings.Region
+	}
+
+	return &Provider{
+		domain:             domain,
+		host:               host,
+		ipVersion:          ipVersion,
+		ipv6Suffix:         ipv6Suffix,
+		awsAccessKeyID:     providerSpecificSettings.AwsAccessKey,
+		awsSecretAccessKey: providerSpecificSettings.AwsSecretAccessKey,
+		hostedZoneID:       providerSpecificSettings.HostedZoneID,
+		region:             region,
+		ttl:                ttl,
+	}, nil
+}
+
+type settings struct {
+	AwsAccessKey       string `json:"aws_access_key"`
+	AwsSecretAccessKey string `json:"aws_secret_access_key"`
+	Region             string `json:"region"`
+	HostedZoneID       string `json:"hosted_zone_id"`
+	TTL                *int64 `json:"ttl"`
+}
+
+func validateSettings(providerSpecificSettings settings, domain, host string) error {
+	// TODO: update this switch to be as restrictive as possible
+	// to fail early for the user. Use errors already defined
+	// in the internal/provider/errors package, or add your own
+	// if really necessary. When returning an error, always use
+	// fmt.Errorf (to enforce the caller to use errors.Is()).
+	switch {
+	case domain == "":
+		return fmt.Errorf("%w", errors.ErrDomainNotSet)
+	case host == "":
+		return fmt.Errorf("%w", errors.ErrHostNotSet)
+	case providerSpecificSettings.AwsAccessKey == "":
+		return fmt.Errorf("%w", errors.ErrAccessKeyIDNotSet)
+	case providerSpecificSettings.AwsSecretAccessKey == "":
+		return fmt.Errorf("%w", errors.ErrSecretAccessKeyNotSet)
+	case providerSpecificSettings.HostedZoneID == "":
+		return fmt.Errorf("%w", errors.ErrHostedZoneIDNotSet)
+	}
+	return nil
+}
+
+func (p *Provider) String() string {
+	return utils.ToString(p.domain, p.host, constants.AWS, p.ipVersion)
+}
+
+func (p *Provider) Domain() string {
+	return p.domain
+}
+
+func (p *Provider) Host() string {
+	return p.host
+}
+
+func (p *Provider) IPVersion() ipversion.IPVersion {
+	return p.ipVersion
+}
+
+func (p *Provider) IPv6Suffix() netip.Prefix {
+	return p.ipv6Suffix
+}
+
+func (p *Provider) Proxied() bool {
+	return false
+}
+
+func (p *Provider) BuildDomainName() string {
+	return utils.BuildDomainName(p.host, p.domain)
+}
+
+func (p *Provider) HTML() models.HTMLRow {
+	return models.HTMLRow{
+		Domain:    fmt.Sprintf("<a href=\"http://%s\">%s</a>", p.BuildDomainName(), p.BuildDomainName()),
+		Host:      p.Host(),
+		Provider:  "<a href=\"https://aws.amazon.com/route53/\">Amazon Route 53</a>",
+		IPVersion: p.ipVersion.String(),
+	}
+}
+
+func (p *Provider) Update(ctx context.Context, client *http.Client, ip netip.Addr) (newIP netip.Addr, err error) {
```

Comment:
Is there an option in the AWS SDK to use the given `client`? This is somehow important for the rest of the program

Reply:
As requested, I removed the SDK dependency, code is a bit bigger now, no need for change anymore I suppose.

Reply:
This is awesome! Congrats! 💯